src/octoprint/plugins/pluginmanager/__init__.py

```diff
@@ -497,6 +511,38 @@ def condition():
             unless=lambda: refresh,
         )(view)()
 
+    @octoprint.plugin.BlueprintPlugin.route("/plugins/versions")
+    @Permissions.PLUGIN_PLUGINMANAGER_LIST.require(403)
+    def retrieve_plugin_list(self):
+        refresh = request.values.get("refresh", "false") in valid_boolean_trues
+        if refresh or not self._is_notices_cache_valid():
+            self._notices_available = self._refresh_notices()
+
+        def view():
+            return jsonify(**self._plugin_versions_response())
+
+        def etag():
+            import hashlib
+
+            hash = hashlib.sha1()
+
+            def hash_update(value):
+                value = value.encode("utf-8")
+                hash.update(value)
+
+            hash_update(repr(self._plugin_versions_response()))
+            hash_update(repr(_DATA_FORMAT_VERSION))
+            return hash.hexdigest()
+
+        def condition():
+            return check_etag(etag())
+
+        return with_revalidation_checking(
```

Comment:
Considering that the full response is required in order to determine the ETag needed for revalidation checking, I wouldn't bother with caching here since it simply doesn't add anything in terms of saving resources.

If I need to fetch all enabled plugins & versions to figure out if they changed in order to respond with a `Not Modified` header, there's not much point in caching to begin with.

I'd rather suggest to get rid of the additional response generating method defined above altogether and inlining that here.

Reply:
So basically we can get rid of the entire ETag/caching and just return a simple json response?

Reply:
IMHO yes. I mean, we don't really gain any performance from caching here, and in worst case (cache is stale) will actually have to do the work needed to preparing the response twice (once for the ETag comparison and once for the actual response generation).

Reply:
Totally agree...I basically just copied it from the other endpoints but guess the situation there is a bit different. Will remove it, also passively resolves some of the other points!